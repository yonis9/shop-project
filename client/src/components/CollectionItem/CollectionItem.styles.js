import styled from 'styled-components';
import CustomButton from '../CustomButton/CustomButton';

export const CollectionItemContainer = styled.div`
    width: 22vw;
    display: flex;
    flex-direction: column;
    height: 350px;
    align-items: center;
    position: relative;

    &:hover {
        .image {
            opacity: 0.8;
          }

        button {
          display: flex;
          opacity: 0.9;
        }
`

export const AddButton = styled(CustomButton)`
  width: 80%;
  opacity: 0.7;
  position: absolute;
  top: 255px;
  display: none;
`;

export const BackgroundImage = styled.div`
    width: 100%;
    height: 95%;
    background-size: cover;
    background-position: center;
    margin-bottom: 5px;
    background-image: url(${(props) => props.imageUrl})
`
export const CollectionFooterContainer = styled.div`
    padding:  0 20px;
    width: 100%;
    height: 5%;
    display: flex;
    justify-content: space-evenly;
    font-size: 18px;
`

export const NameContainer = styled.span`
    width: 90%;
    margin-bottom: 15px;
`

export const PriceContainer = styled.span`
    width: 10%;
    text-align: right;
`



