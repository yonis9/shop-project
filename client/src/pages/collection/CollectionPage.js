import React from 'react';
import { connect } from 'react-redux';

import { collectionSelector } from '../../redux/shop/shop-selectors';

import CollectionItem from '../../components/CollectionItem/CollectionItem';

import {
    ColleactionPageContainer,
    CollectionTitle,
    CollectionItemsContainer
} from './CollectionPage.styles';

export const CollectionPage = ({ collection }) => {
   const { title, items } = collection

    return (
    <ColleactionPageContainer>
        <CollectionTitle>{title}</CollectionTitle>
        <CollectionItemsContainer>
            {
                items.map(item => <CollectionItem key={item.id} item={item} />)
            }
        </CollectionItemsContainer>
    </ColleactionPageContainer>

)}

const MapStateToProps = (state, ownProps) => ({
    collection: collectionSelector(ownProps.match.params.collectionId)(state)
})

export default connect(MapStateToProps)(CollectionPage);