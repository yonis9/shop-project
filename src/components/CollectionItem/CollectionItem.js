import React from 'react';

import CustomButton from '../CustomButton/CustomButton';

import './CollectionItem.scss';

const CollectionItem = ( { name, imageUrl, price }) => (
    <div className='collection-item'>
        <div className='image'
             style={{backgroundImage: `url(${imageUrl})`}}/>
        <div className='collection-footer'>
            <span className='name'>{name}</span>
            <span className='price'>${price}</span>
        </div>
        <CustomButton inverted>ADD TO CART</CustomButton>
    </div>
)

export default CollectionItem