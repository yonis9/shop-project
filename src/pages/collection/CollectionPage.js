import React from 'react';
import { connect } from 'react-redux';

import { collectionSelector } from '../../redux/shop/shop-selectors';

import CollectionItem from '../../components/CollectionItem/CollectionItem';

import './CollectionPage.scss';

const CollectionPage = ({ collection }) => {
   const { items } = collection

    return (
    <div className='collection-page'>
        <div className='title'>{collection.title}</div>
        <div className='items'>
            {
                items.map(item => <CollectionItem key={item.id} item={item} />)
            }
        </div>
    </div>

)}

const MapStateToProps = (state, ownProps) => ({
    collection: collectionSelector(ownProps.match.params.collectionId)(state)
})

export default connect(MapStateToProps)(CollectionPage);