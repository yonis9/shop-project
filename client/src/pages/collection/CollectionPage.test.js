import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPage } from './CollectionPage';
import CollectionItem from '../../components/CollectionItem/CollectionItem';

describe('CollectionPage', () => {
    let wrapper;
    let mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }];

    beforeEach(() => {
        const mockCollection = {
            title: 'men',
            items: mockItems
        };

        wrapper = shallow(<CollectionPage collection={mockCollection}/>);
    });

    it('should render CollectionPage', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the same number of CollectionItems as collection array length', () => {
        expect(wrapper.find(CollectionItem).length).toBe(mockItems.length)
    });
});