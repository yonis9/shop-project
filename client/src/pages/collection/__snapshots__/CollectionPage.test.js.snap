// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage should render CollectionPage 1`] = `ShallowWrapper {}`;
